src/common/MultipleFilesDialog.tsx

```diff
@@ -37,6 +38,15 @@ export const MultipleFilesDialog = ({ callback }: MultipleFilesDialogProps) => {
 };
 
 const MultipleFilesDialogBody = () => {
+  const fs = useFileSystem();
+  const [fileCount, setFileCount] = useState<number>(1);
+  useEffect(() => {
+    const getNumFiles = async () => {
+      const result = (await fs.statistics()).files;
+      setFileCount(result);
+    };
+    getNumFiles();
+  }, [fs, setFileCount]);
```

Comment:
I don't see how this would update as required. It kinda works but only because the dialog is a modal so it works to fetch it once but it seems a fragile pattern (fs and setFileCount will never change identity).

Please use the useProject() hook and project.files.length.

Reply:
Fixed here and where the same pattern was used elsewhere - [`e958238`](https://github.com/microbit-foundation/python-editor-next/pull/828/commits/e958238a62f19c96cd967a0e73d023b27a0f35a9).

Reply:
Thanks.